middleware/src/middleware.go

```diff
@@ -521,38 +667,105 @@ func (middleware *Middleware) SetRootPassword(args rpcmessages.SetRootPasswordAr
+				Message: strings.Join(out, "\n"),
+				Code:    errorCode,
+			}
 		}
+
 		return rpcmessages.ErrorResponse{Success: true}
 	}
-	return rpcmessages.ErrorResponse{Success: false, Message: "invalid password"}
+
+	return rpcmessages.ErrorResponse{
+		Success: false,
+		Message: "The password has to be at least 8 chars. An unicode char is counted as one.",
+		Code:    rpcmessages.ErrorSetRootPasswordTooShort,
+	}
 }
 
 // runBBBCmdScript runs the bbb-cmd.sh script.
 // The script executes commands like for example mounting a USB drive, doing a backup and copying files.
-func (middleware *Middleware) runBBBCmdScript(method string, arg1 string, arg2 string) (out []byte, err error) {
-	script := middleware.environment.GetBBBCmdScript()
-	cmdAsString := strings.Join([]string{script, method, arg1, arg2}, " ")
-	out, err = exec.Command(script, method, arg1, arg2).Output()
-	if err != nil {
-		// no error handling here, only logging.
-		log.Printf("Error: The command '%s' exited with the output '%v' and error '%s'.\n", cmdAsString, string(out), err.Error())
-	}
+func (middleware *Middleware) runBBBCmdScript(args []string) (outputLines []string, err error) {
+	outputLines, err = runCommand(middleware.environment.GetBBBCmdScript(), args)
 	return
 }
 
 // runBBBConfigScript runs the bbb-config.sh script.
 // The script changes the system configuration in redis by setting or unsetting the appropriate keys.
 // If necessary the affected services are restarted.
-func (middleware *Middleware) runBBBConfigScript(method string, arg1 string, arg2 string) (out []byte, err error) {
-	script := middleware.environment.GetBBBConfigScript()
-	cmdAsString := strings.Join([]string{script, method, arg1, arg2}, " ")
-	out, err = exec.Command(script, method, arg1, arg2).Output()
+func (middleware *Middleware) runBBBConfigScript(args []string) (outputLines []string, err error) {
+	outputLines, err = runCommand(middleware.environment.GetBBBConfigScript(), args)
+	return
+}
+
+// runCommand runs the passed command and returns the combined stdout and stderr output.
+// If the command could not be run, err is not nil.
+func runCommand(command string, args []string) (combinedLines []string, err error) {
+	cmd := exec.Command(command, args...)
+	log.Printf("executing command: %s %s", command, strings.Join(args, " "))
+
+	rawstdoutStderr, err := cmd.CombinedOutput()
 	if err != nil {
-		// no error handling here, only logging.
-		log.Printf("Error: The command '%s' exited with the output '%v' and error '%s'.\n", cmdAsString, string(out), err.Error())
+		// no error handling here, just logging
+		log.Printf("Error executing command '%s %s': '%s'", command, strings.Join(args, " "), err.Error())
 	}
-	return
+
+	combined := strings.TrimSuffix(string(rawstdoutStderr), "\n")
+	combinedLines = strings.Split(combined, "\n")
+	return combinedLines, err
+}
+
+func handleBBBScriptError(outputLines []string, err error, possibleErrors []rpcmessages.ErrorCode) rpcmessages.ErrorCode {
```

Comment:
Oh, could you also rename `handleBBBScriptError()` to something more like `handleBBBScriptErrorCode()` since it's just for returning `ErrorCode`